src/test/java/edu/oakland/test/database/DatabasePersistentStorageTest.java

```diff
@@ -0,0 +1,326 @@
+package edu.oakland.test.database;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import edu.oakland.helper.admin.LocationDataPoint;
+import edu.oakland.helper.admin.TrackData;
+import edu.oakland.production.database.DatabasePersistentStorage;
+import edu.oakland.production.database.DatabasePersistentStorageClass;
+import java.lang.IllegalArgumentException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Random;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+@DisplayName("Database PersistentStorage Test")
+public class DatabasePersistentStorageTest {
+
+  String[] randomArray = {"A", "B"};
+
+  LocationDataPoint ldp = new LocationDataPoint(0, 0, LocalDateTime.of(
+        (int) (Math.random() * 50 + 1970),
+        (int) (Math.random() * 12 + 1),
+        (int) (Math.random() * 28 + 1),
+        (int) (Math.random() * 24),
+        (int) (Math.random() * 60)
+    ));
+
+
+  float course = generateRandomCourse();
+  TrackData trackDataPoint = new TrackData(
+        generateRandomLocationDataPointsArray(5),
+        course,
+        generateRandomSpeed()
+    );
+
+
+  ArrayList<LocationDataPoint> locDataPoint = new ArrayList<LocationDataPoint>();
+  ArrayList<TrackData> trackData = new ArrayList<TrackData>();
+
+  
+
+  @Test
+  @DisplayName("RFID data in is RFID data out")
+  void rfidDataInIsRfidDataOut() {
+    //int random = (int) (Math.random() * 60);
```

Comment:
I would go back to the random generation here, it makes the testing much more robust and reliable.
```suggestion
    int random = (int) (Math.random() * 60);
```